### PR TITLE
fix doxygen for sorc/emcsfc_ice_blend.fd (#342)

### PR DESCRIPTION
**Pull request by lgannoaa**
doxygen update

### CODE REVIEW

File: sorc/emcsfc_ice_blend.fd/emcsfc_ice_blend.f90 [hunk 1]
```diff
@@ -86,6 +87,17 @@
 !!
 !! In the SH, the blended value is simply the 5-minute ice concentration
 !! at 'water' points.  'Coast' and 'land' points are bitmapped out.
+!!
+!! @param imax, jmax constant value on i and j dim
```
Comment (edwardhartnett) [inline sorc/emcsfc_ice_blend.fd/emcsfc_ice_blend.f90:91]: Well these are not parameters. Only functions and subroutines have paramters.

These document variables and we can use the !< in the code to add them to the variable declaration line of code.

For example, change the code which is now:

```
integer, parameter     :: imax=4320
integer, parameter     :: jmax=2160

```
to:

```
integer, parameter     :: imax=4320 !< Constant value on i dim.
integer, parameter     :: jmax=2160 !< Constant value on j dim.

```

And remove all the @param lines.

**Review by edwardhartnett — CHANGES_REQUESTED** [review]

File: sorc/emcsfc_ice_blend.fd/emcsfc_ice_blend.f90 (hunk 1 above)
Comment (lgannoaa) [reply sorc/emcsfc_ice_blend.fd/emcsfc_ice_blend.f90]: Done.

File: sorc/emcsfc_ice_blend.fd/emcsfc_ice_blend.f90 [hunk 2]
```diff
@@ -97,8 +98,8 @@ program emcsfc_ice_blend
 
  character(len=200)     :: infile, outfile
 
- integer, parameter     :: imax=4320
- integer, parameter     :: jmax=2160
+ integer, parameter     :: imax=4320 !< Constant value on i dim.
+ integer, parameter     :: jmax=2160 !< Constant value on j dim.
```
Comment (edwardhartnett) [inline sorc/emcsfc_ice_blend.fd/emcsfc_ice_blend.f90:102]: You need to transfer all the documentation. So the descriptions of the other vars, not just imax and jmax.

**Review by edwardhartnett — CHANGES_REQUESTED** [review]

File: sorc/emcsfc_ice_blend.fd/emcsfc_ice_blend.f90 (hunk 2 above)
Comment (lgannoaa) [reply sorc/emcsfc_ice_blend.fd/emcsfc_ice_blend.f90:102]: How can I transfer documentation if there is no existing to work with? Please give me an example using emcsfc_ice_blend.f90.

**Comment by edwardhartnett** [comment]
!! @param j     search position of a file
￼!! @param lugi  set to 0 - no grib index file
￼!! @param jdisc set to 2 - search for discipline
￼!! @param jpdtn search for product definition template number
￼!! @param jgdtn search for grid definition template number; 0 - lat/lon grid
￼!! @param jids  array of values in identification section, set to wildcard
￼!! @param jgdt  array of values in grid definition template 3.m
￼!! @param jpdt  array of values in product definition template 4.n
￼!! @param unpack switch for unpack data

**Comment by lgannoaa** [comment]
Updated from your suggestion. Please check it again.

File: sorc/emcsfc_ice_blend.fd/emcsfc_ice_blend.f90 [hunk 3]
```diff
@@ -87,6 +87,15 @@
...
 !! In the SH, the blended value is simply the 5-minute ice concentration
 !! at 'water' points.  'Coast' and 'land' points are bitmapped out.
+!! @param j search position of a file
+!! @param lugi set to 0 - no grib index file
+!! @param jdisc set to 2 - search for discipline
+!! @param jpdtn search for product definition template number
+!! @param jgdtn search for grid definition template number; 0 - lat/lon grid
+!! @param jids array of values in identification section, set to wildcard
+!! @param jgdt array of values in grid definition template 3.m
+!! @param jpdt array of values in product definition template 4.n
```
Comment (edwardhartnett) [inline sorc/emcsfc_ice_blend.fd/emcsfc_ice_blend.f90:97]: Well, we're back to where we started. ;-)

These values are not parameters so they should not be documented with the @param tag.

Instead, these comments should be applied to the variables defined below.

Do that by using the '!<' doxygen comment marker, like this:

`integer :: some_variable !< Description of that variable.
`

So move the comments to the variables, and get rid of the param statements.

Comment (lgannoaa) [reply sorc/emcsfc_ice_blend.fd/emcsfc_ice_blend.f90]: I understood from your reply above as your example. I modified it again hope to in-line with yours this time.

File: sorc/emcsfc_ice_blend.fd/emcsfc_ice_blend.f90 [hunk 4]
```diff
@@ -110,15 +101,21 @@ program emcsfc_ice_blend
  integer, parameter     :: imax=4320 !< Constant value on i dim.
  integer, parameter     :: jmax=2160 !< Constant value on j dim.
 
- integer                :: i,j, istat, iunit
+ integer                :: i,j, istat, iunit !< j search position of a file
  integer                :: ii, iii, jj, jjj, count
- integer                :: lugi
- integer                :: jdisc, jgdtn, jpdtn, k
+ integer                :: lugi !< set to 0 - no grib index file
+ integer                :: jdisc, jgdtn, jpdtn, k
```
Comment (edwardhartnett) [inline sorc/emcsfc_ice_blend.fd/emcsfc_ice_blend.f90:107]: You need to break the declaration of each var onto it's own line.

```
integer :: jdisk !< set to 2 - search for discipline
integer:: jgdtn !<  search for product definition template number
integer::jptn !< search for grid definition template number; 0 - lat/lon grid
integer:: k !< index
```

Comment (lgannoaa) [reply sorc/emcsfc_ice_blend.fd/emcsfc_ice_blend.f90]: Updated

File: sorc/emcsfc_ice_blend.fd/emcsfc_ice_blend.f90 [hunk 5]
```diff
@@ -197,11 +195,11 @@ program emcsfc_ice_blend
  do j = 1, jmax
  do i = 1, imax
    if (dummy(i,j) < 0.1) then
-      mask_5min(i,j)=0   ! water
+      mask_5min(i,j)=0   !< water
```
Comment (edwardhartnett) [inline sorc/emcsfc_ice_blend.fd/emcsfc_ice_blend.f90:198]: No, other than variable declarations, no doxygen is needed within the functions.

Comment (lgannoaa) [reply sorc/emcsfc_ice_blend.fd/emcsfc_ice_blend.f90]: removed.

**Review by edwardhartnett — APPROVED** [review]

**Comment by lgannoaa** [comment]
Thank you

**Comment by edwardhartnett** [comment]
Why did you close the PR? I have not merged it yet. ;-)

I am waiting for @GeorgeGayno-NOAA to be finished with #238 before merging this.

Can you undelete your branch?

**Pull request closed without merging**